### PR TITLE
[AIRFLOW-2355] parameters passed through trigger_dag -c should be available in subdag

### PR DESCRIPTION
**Pull request by milton0825**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2355
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Parameters passed through `airflow trigger_dag -c '{"text": "blah"}'` can be accessed through 
`{{ dag_run.conf.text }}` in subdag.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by milton0825** [comment]
PTAL @Fokko @mistercrunch @kaxil 
TravisCI passed: https://travis-ci.org/milton0825/incubator-airflow/builds/389828444

**Comment by milton0825** [comment]
PTAL @Fokko @jgao54 @kaxil

File: airflow/api/common/experimental/trigger_dag.py [hunk 1]
```diff
@@ -25,14 +25,19 @@
 from airflow.utils.state import State
 
 
-def trigger_dag(dag_id, run_id=None, conf=None, execution_date=None,
-                replace_microseconds=True):
-    dagbag = DagBag()
-
-    if dag_id not in dagbag.dags:
+def _trigger_dag(
```
Comment (kaxil) [inline airflow/api/common/experimental/trigger_dag.py:28]: I don't see a need of new function here. Would it not be possible to add this code to `trigger_dag` itself?

Comment (milton0825) [reply airflow/api/common/experimental/trigger_dag.py:28]: Well it makes it easier for testing as we can inject mocked `dag_run`, `dag_bag`, and `dag`.

**Review by kaxil — APPROVED** [review]
LGTM

**Pull request closed without merging**